src/main/js/BranchWPsForm.tsx

```diff
@@ -80,7 +80,6 @@ class BranchWPsForm extends React.Component<Props, State> {
   userBranchPermissionAdded = (permission: BranchWP) => {
     this.setState(
       {
-        ...this.state,
         permissions: [...this.state.permissions, permission]
```

Comment:
The state of the permission is not preserved. If i select DENY and add the permission, it is reseted back to ALLOW after creation.

Reply:
Changed this.